packages/nbdime/src/common/mergeview.ts

```diff
@@ -143,563 +341,404 @@ function createNbdimeMergeView(
+    this.clearHighlighting(
+      remoteEditor,
+      this.model.additions,
+      this.chunks,
+      DIFF_OP.DIFF_INSERT
+    );
+
+    this.clearHighlighting(
+      baseEditor,
+      this.model.deletions,
+      this.chunks,
+      DIFF_OP.DIFF_DELETE
+    );
+
+    this.updateHighlighting(
+      remoteEditor,
+      this.model.additions,
+      this.chunks,
+      DIFF_OP.DIFF_INSERT
+    );
+
+    this.updateHighlighting(
+      baseEditor,
+      this.model.deletions,
+      this.chunks,
+      DIFF_OP.DIFF_DELETE
+    );
+
+    this.syncScroll(baseEditor, remoteEditor);
+    this.syncScroll(remoteEditor, baseEditor);
   }
-
-  syncModel() {
-    if (this.modelInvalid()) {
-      let edit = this.ownEditor;
+    syncModel() {
+      if (!this.modelInvalid()) {
+        return;
+      }
+      let editor = this.remoteEditorWidget.cm;
       let updatedLineChunks = this.model.getLineChunks();
       let updatedChunks = lineToNormalChunks(updatedLineChunks);
-      if (this.model.remote === edit.getValue()) {
+      if (this.model.remote === editor.state.doc.toString()) {
         // Nothing to do except update chunks
         this.lineChunks = updatedLineChunks;
         this.chunks = updatedChunks;
         return;
       }
-      let cursor = edit.getDoc().getCursor();
+      let cursor = editor.state.selection.main.head
       let newLines = splitLines(this.model.remote!);
-      let start = edit.getDoc().firstLine();
-      let last = edit.getDoc().lastLine() + 1;
-      let cumulativeOffset = 0;
-      let end: number;
-      let updatedEnd: number;
-      // We want to replace contents of editor, but if we have collapsed regions
-      // some lines have been optimized away. Carefully replace the relevant bits:
-      for (let range of this.collapsedRanges) {
-        let baseLine = range.line;
-        end = getMatchingEditLine(baseLine, this.chunks);
-        updatedEnd = getMatchingEditLine(baseLine, updatedChunks);
-        let offset = updatedEnd - end;
-        if (end !== start || offset !== 0) {
-          edit.getDoc().replaceRange(
-            newLines.slice(start + cumulativeOffset, updatedEnd + cumulativeOffset - 1).join(''),
-            CodeMirror.Pos(start, 0),
-            CodeMirror.Pos(end - 1, 0),
-            'syncModel'
-          );
-        }
-        cumulativeOffset += offset;
-        start = end + range.size;
-      }
-      if (start < last) {
-        // Only here if no collapsed ranges, replace full contents
-        edit.getDoc().replaceRange(
-          newLines.slice(start, newLines.length).join(''),
-          CodeMirror.Pos(start, 0),
-          CodeMirror.Pos(last, 0),
-          'syncModel'
-        );
-      }
-      this.ownEditor.getDoc().setCursor(cursor);
+      editor.dispatch({
+        changes: {from: 0, to: editor.state.doc.length, insert: newLines.slice(0, newLines.length).join('')}
+      });
+      this.remoteEditorWidget.cm.dispatch({selection: {anchor: cursor}})
```

Comment:
Nitpicking: use editor instead.
I think you can dispatch the changes and the selection in a single call

Reply:
Thanks. This has been taken into account.